packages/open-next/src/build/copyTracedFiles.ts

```diff
@@ -0,0 +1,256 @@
+
+  const extractFiles = (files: string[], from = standaloneNextDir) => {
+    return files.map((f) => path.resolve(from, f));
+  };
+
+  // On next 14+, we might not have to include those files
+  // For next 13, we need to include them otherwise we get runtime error
+  const requiredServerFiles = JSON.parse(
+    readFileSync(
+      path.join(
+        dotNextDir,
+        bundledNextServer
+          ? "next-minimal-server.js.nft.json"
+          : "next-server.js.nft.json",
+      ),
+      "utf8",
+    ),
+  );
+
+  const filesToCopy = new Map<string, string>();
+
+  // Files necessary by the server
+  extractFiles(requiredServerFiles.files).forEach((f) => {
+    filesToCopy.set(f, f.replace(standaloneDir, outputDir));
+  });
+
+  // create directory for pages
+  if (existsSync(path.join(standaloneDir, ".next/server/pages"))) {
+    mkdirSync(path.join(outputNextDir, "server/pages"), {
+      recursive: true,
+    });
+  }
+  if (existsSync(path.join(standaloneDir, ".next/server/app"))) {
+    mkdirSync(path.join(outputNextDir, "server/app"), {
+      recursive: true,
+    });
+  }
+
+  mkdirSync(path.join(outputNextDir, "server/chunks"), {
+    recursive: true,
+  });
+
+  const computeCopyFilesForPage = (pagePath: string) => {
+    const fullFilePath = `server/${pagePath}.js`;
+    let requiredFiles;
+    try {
+      requiredFiles = JSON.parse(
+        readFileSync(
+          path.join(standaloneNextDir, `${fullFilePath}.nft.json`),
+          "utf8",
+        ),
+      );
+    } catch (e) {
+      //TODO: add a link to the docs
+      throw new Error(
+        `
+--------------------------------------------------------------------------------
+${pagePath} cannot use the edge runtime.
+OpenNext requires edge runtime function to be defined in a separate function. 
+See the docs for more information on how to bundle edge runtime functions.
+--------------------------------------------------------------------------------
+        `,
+      );
+    }
+    const dir = path.dirname(fullFilePath);
+    extractFiles(
+      requiredFiles.files,
+      path.join(standaloneNextDir, dir),
+    ).forEach((f) => {
+      filesToCopy.set(f, f.replace(standaloneDir, outputDir));
+    });
+
+    filesToCopy.set(
+      path.join(standaloneNextDir, fullFilePath),
+      path.join(outputNextDir, fullFilePath),
+    );
+  };
+
+  const hasPageDir = routes.some((route) => route.startsWith("pages/"));
+  const hasAppDir = routes.some((route) => route.startsWith("app/"));
+
+  // We need to copy all the base files like _app, _document, _error, etc
+  // One thing to note, is that next try to load every routes that might be needed in advance
+  // So if you have a [slug].tsx at the root, this route will always be loaded for 1st level request
+  // along with _app and _document
+  if (hasPageDir) {
+    //Page dir
+    computeCopyFilesForPage("pages/_app");
+    computeCopyFilesForPage("pages/_document");
+    computeCopyFilesForPage("pages/_error");
```

Comment:
Nice catch,
Yes we do, for the created functions, we only copy the page in `route`.
We should probably add this inside a try catch as they are usually static or SSG and in those cases, we might not have a `nft.json` file

Reply:
We can also make `computeCopyFilesForPage` async, then use `await Promise.all([computeCopyFilesForPage...])`
to run the `fs` operations in parallel to speed up the build time.

Might be worth considering if the time save is significant.

Reply:
Yeah there is some room for improvement in terms of performance here. But `computeCopyFilesForPage` doesn't actually copy anything, it just compute every file that is necessary for a given server function.
The copying is done later and that's probably where it would help us.

At the moment it is fast enough i'd say (A lot more than the actual `next build`)

We could even improve the runtime performance if we do this properly.
https://github.com/sst/open-next/blob/dae9d98b0519c1ecb640b8c80d7fe0acca5b3886/packages/open-next/src/build/copyTracedFiles.ts#L189-L195

Reply:
I added the 404 and 500, they can be there if the user uses `getStaticProps` in them.

As for the performance improvement, i think we should see later if it becomes an issue. Making everything async may cause some other trouble as we may be copying thousands of files at the same time